### PR TITLE
Create product

### PR DESCRIPTION
**Pull request by luisoliveira98**

### CODE REVIEW

File: src/main/java/com/tqs/plazzamarket/controllers/ProductController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        product.setDescription(productJson.get("description").toString());
+        product.setPrice(Double.parseDouble(productJson.get("price").toString()));
+        product.setQuantity(Double.parseDouble(productJson.get("quantity").toString()));
+        if (productJson.containsKey("category"))
+            product.setCategory(categoryRepository.getOne(productJson.get("category").toString()));
+        System.out.println(product);
+        return new ResponseEntity<>(productRepository.save(product), HttpStatus.CREATED);
+    }
+
+    @GetMapping(path = "/products/")
```
Comment (lengors) [inline src/main/java/com/tqs/plazzamarket/controllers/ProductController.java]: Add To A different branch

File: src/main/java/com/tqs/plazzamarket/controllers/CategoryController.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+
+    @RequestMapping(path = "/category/init", method = RequestMethod.GET)
+    public void init() {
+        System.out.println("Aqui");
+        Category c = new Category("Flowers");
+        categoryRepository.save(c);
+    }
+
+    @GetMapping(path = "/categories/")
```
Comment (lengors) [inline src/main/java/com/tqs/plazzamarket/controllers/CategoryController.java]: Add to a different branch

File: src/test/java/com/tqs/plazzamarket/controllers/ProductControllerIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    public void testCreateProduct() throws Exception{
+        Map<String, Object> productJSON = new HashMap<>();
+        productJSON.put("name", "Potato");
+        productJSON.put("quantity", "4");
+        productJSON.put("price", "5");
+        productJSON.put("description", "test");
+
+        System.out.println(mapper.writeValueAsString(productJSON));
+
+        mvc
```
Comment (lengors) [inline src/test/java/com/tqs/plazzamarket/controllers/ProductControllerIntegrationTest.java]: Verify results returned

File: src/test/java/com/tqs/plazzamarket/controllers/ProductControllerIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            .perform(MockMvcRequestBuilders.post("/api/products/add").contentType(MediaType.APPLICATION_JSON)
+                    .content(mapper.writeValueAsString(productJSON)))
+            .andExpect(MockMvcResultMatchers.status().isCreated());
+
+        //Optional<Product> optional = productRepository.findById(0);
```
Comment (lengors) [inline src/test/java/com/tqs/plazzamarket/controllers/ProductControllerIntegrationTest.java]: Add repository check

File: src/test/java/com/tqs/plazzamarket/controllers/ProductControllerUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        productJSON.put("description", "test");
+
+        Product p = new Product();
+        p.setName("Potato");
+        p.setQuantity(4);
+        p.setPrice(5);
+        p.setDescription("test");
+        BDDMockito.given(productRepository.saveAndFlush(p)).willReturn(p);
+
+        mvc
```
Comment (lengors) [inline src/test/java/com/tqs/plazzamarket/controllers/ProductControllerUnitTest.java:54]: Verify results returned

File: src/test/java/com/tqs/plazzamarket/repositories/CategoryRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
+package com.tqs.plazzamarket.repositories;
```
Comment (lengors) [inline src/test/java/com/tqs/plazzamarket/repositories/CategoryRepositoryTest.java:1]: Add test to a different branch

**Review by lengors — CHANGES_REQUESTED** [review]

**Review by nunosilva12345 — APPROVED** [review]
Done without errors !
Works fine

**Pull request merged**